### ISSUE TITLE
Unable to implement "Hello_world" example on esp32

### ISSUE THREAD

**Issue by kunalkul** [tensorflow/tensorflow#37007]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution : Windows 10
- TensorFlow installed from : installed on esp-idf using `pip install tensorflow`
- Tensorflow version (commit SHA if source):
- Target platform : esp32-devKitC-V4

**Describe the problem**
I have followed [this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/hello_world/README.md#deploy-to-esp32) guide step-by-step so far
I managed to install the esp-idf and also checked that the env. variables are in PATH (following the guide) but unable to proceed further

**Exact sequence of commands/steps when you ran into the problem**
after doing `pip install tensorflow` (on esp-idf cmd prompt), I entered this command:
`make -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project`

and the output is this:
`'make' is not recognized as an internal or external command,
operable program or batch file.`

I also tried this command:
`cmake -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project`

and this was the error:
`CMake Error: The source directory "C:/Users/kkulk/Desktop/esp-idf/generate_hello_world_esp_project" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.`

**I am a complete newbie to esp32, tensorflow and github as well, please let me know what the problem seems to be and if I am overlooking something completely obvious or did something wrong**

**Comment by vikramdattu** [comment]
@kunalkul Actually, there is an issue (which is not an issue!) with TFMicro project gen system.

When you generate an example and try to generate different example after that it generates it successfully and throws the error message you get.

For now, please ignore the message and go ahead building the example generated.

**Comment by AdityaHPatwardhan** [comment]
@kunalkul , It seems you have installed tensorflow with `pip install tensorflow`, that should be done when you want to use tensorflow functions python code.
For being able to compile the [tensorflow/lite/micro/examples](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/micro/examples) and flash the binaries on the respective devices, you need to download the source code as mentioned [here](https://www.tensorflow.org/install/source_windows#download_the_tensorflow_source_code)

Also,
>  'make' is not recognized as an internal or external command, operable program or batch file.

it seems you have not installed make utility on your machine , maybe [this](http://gnuwin32.sourceforge.net/packages/make.htm) could help.

In short you should proceed as follows
1) Install source code of [tensorflow](https://github.com/tensorflow/tensorflow) ( clone this repository) as mentioned above
2)  Install make if not already installed 
3) Note: When you successfully install all the required dependancies and try to build hello-world example with following command
> make -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project

Please note that it might print an error message, as currently there seems to be a build issuer which needs a fix, but even with the error message the build actually succeeds and you can go to example folder and compile the binary for ESP with ‘idf.py build’.

**Comment by kunalkul** [comment]
I executed steps 1 and 2, and ran

`make -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project`

got the following error : `C:\Users\kkulk\tensorflow\tensorflow\lite\micro\tools\make\Makefile:2: *** "Requires make version 3.82 or later (current is 3.81)".  Stop.`

I proceeded ahead by running `idf.py build`
and got this error : `Checking Python dependencies...
Python requirements from C:\Users\kkulk\Desktop\esp-idf\requirements.txt are satisfied.
Executing action: all (aliases: build)
CMakeLists.txt not found in project directory c:\users\kkulk\tensorflow\tensorflow\lite\micro\examples\hello_world`

**Comment by AdityaHPatwardhan** [comment]
@kunalkul The errors are arising because of faulty setup of tensorflow. Please do not go to next step unless previous step is completed as it is surely not gonna work.
Plase see that you have successfully cloned [tensorflow](https://github.com/tensorflow/tensorflow/). 
please change you have make version > 3.81 as recommended.
If you get errors such as cannot find file, then it seems you have not cloned the repository correctly

**Comment by kunalkul** [comment]
I was able to clone tensorflow succesfully, as I am able to navigate to the directory on my PC both through file explorer and cmd (This was after I updated my make version to 4.3, as I have already mentioned in the **edit** comment above).

I am not sure what's causing this error : `process_begin: CreateProcess(NULL, uname -m, ...) failed. tensorflow/lite/micro/tools/make/Makefile:25: pipe: No error -m was unexpected at this time. tensorflow/lite/micro/tools/make/download_and_extract.sh "https://github.com/google/gemmlowp/archive/719139ce755a0f31cbf1c37f7f98adcc7fc9f425.zip" "7e8191b24853d75de2af87622ad293ba" tensorflow/lite/micro/tools/make/downloads/gemmlowp process_begin: CreateProcess(NULL, bash C:\Users\kkulk\tensorflow\tensorflow\lite\micro\tools\make\download_and_extract.sh https://github.com/google/gemmlowp/archive/719139ce755a0f31cbf1c37f7f98adcc7fc9f425.zip 7e8191b24853d75de2af87622ad293ba tensorflow/lite/micro/tools/make/downloads/gemmlowp, ...) failed. make (e=2): The system cannot find the file specified. make: *** [tensorflow/lite/micro/tools/make/Makefile:261: tensorflow/lite/micro/tools/make/downloads/gemmlowp] Error 2`

**Comment by AdityaHPatwardhan** [comment]
I assume you have setup all the things on windows correctly as mentioned [here](https://www.tensorflow.org/install/source_windows#setup_for_windows), If yes and still you have the problem then maybe you could look into [this issue](https://github.com/dariomanesku/cmft/issues/28), I personally do not use windows for development so if that is an issue related with windows I would not be the right person to comment, Maybe looking at other build issues in [tensorflow/issues](https://github.com/tensorflow/tensorflow/issues) might help.

**Comment by mazurick** [comment]
I am having this issue too. I am going insane. I've installed make but the default is set to 3.81 and upgrading it to 4.1

**Comment by TobiasBoeschMiele** [comment]
Hello everyone,

I would also like to learn to know tensorflow and therefore I'm trying to use the Hello World example.
That fails with errors either in MinGW or cygwin. I have two output lines that express their opinion:

Here comes MinGW:
```
C:\Versionsverwaltung\tensorflow>C:\MinGW\bin\mingw32-make.exe -v
GNU Make 3.82.90
Built for i686-pc-mingw32
Copyright (C) 1988-2012 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

C:\Versionsverwaltung\tensorflow>C:\MinGW\bin\mingw32-make.exe -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project
process_begin: CreateProcess(NULL, uname -m, ...) failed.
"-m" kann syntaktisch an dieser Stelle nicht verarbeitet werden.
FIND: Parameterformat falsch
mingw32-make: *** No rule to make target 'generate_hello_world_esp_project'.  Stop.

C:\Versionsverwaltung\tensorflow>
```


This is cygwin:
```
xxx@XXX/cygdrive/c/Versionsverwaltung/tensorflow
$ make -v
GNU Make 4.3
Gebaut für x86_64-pc-cygwin
Copyright (C) 1988-2020 Free Software Foundation, Inc.
Lizenz GPLv3+: GNU GPL Version 3 oder später <http://gnu.org/licenses/gpl.html>
Dies ist freie Software: Sie können sie nach Belieben ändern und weiter verteilen.
Soweit es die Gesetze erlauben gibt es KEINE GARANTIE.

xxx@XXX/cygdrive/c/Versionsverwaltung/tensorflow
$ make -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project
tensorflow/lite/micro/tools/make/Makefile:297: Warnung: Die Befehle für das Ziel „tensorflow/lite/micro/tools/make/downloads/ruy“ werden überschrieben
tensorflow/lite/micro/tools/make/Makefile:297: Warnung: Alte Befehle für das Ziel „tensorflow/lite/micro/tools/make/downloads/ruy“ werden ignoriert
tensorflow/lite/micro/tools/make/Makefile:297: Warnung: Die Befehle für das Ziel „tensorflow/lite/micro/tools/make/downloads/person_model_grayscale“ werden überschrieben
tensorflow/lite/micro/tools/make/Makefile:297: Warnung: Alte Befehle für das Ziel „tensorflow/lite/micro/tools/make/downloads/person_model_grayscale“ werden ignoriert
tensorflow/lite/micro/tools/make/download_and_extract.sh "https://github.com/google/gemmlowp/archive/719139ce755a0f31cbf1c37f7f98adcc7fc9f425.zip" "7e8191b24853d75de2af87622ad293ba" tensorflow/lite/micro/tools/make/downloads/gemmlowp
tensorflow/lite/micro/tools/make/download_and_extract.sh: Zeile 16: $'\r': Kommando nicht gefunden.
tensorflow/lite/micro/tools/make/download_and_extract.sh: Zeile 24: $'\r': Kommando nicht gefunden.
: Ungültige Optioncro/tools/make/download_and_extract.sh: Zeile 25: set: -
set: Aufruf: set [-abefhkmnptuvxBCHP] [-o Option] [--] [Argument ...]
tensorflow/lite/micro/tools/make/download_and_extract.sh: Zeile 26: $'\r': Kommando nicht gefunden.
tensorflow/lite/micro/tools/make/download_and_extract.sh: Zeile 28: Syntaxfehler beim unerwarteten Wort `$'{\r''
'ensorflow/lite/micro/tools/make/download_and_extract.sh: Zeile 28: `patch_am_sdk() {
make: *** [tensorflow/lite/micro/tools/make/Makefile:297: tensorflow/lite/micro/tools/make/downloads/gemmlowp] Fehler 2

xxx@XXX/cygdrive/c/Versionsverwaltung/tensorflow
```

**Comment by vikramdattu** [comment]
Hi @TobiasBoeschMiele this is because you have Windows style line ending.
Can you please run following command and try again:
`dos2unix tensorflow/lite/micro/tools/make/download_and_extract.sh`

**Comment by TobiasBoeschMiele** [comment]
Thank @vikramdattu for your suggestion.

_I use tensorflow lite at master at https://github.com/tensorflow/tensorflow/commit/89bd08149c9d179bef02a5baf866a20b2fc71fc1_

MinGW has the dos2unix program so I used that. Here is what it puts out when failling again:
```
C:\Versionsverwaltung\tensorflow>C:\MinGW\bin\mingw32-make.exe -v
GNU Make 3.82.90
Built for i686-pc-mingw32
Copyright (C) 1988-2012 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

C:\Versionsverwaltung\tensorflow>C:\MinGW\msys\1.0\bin\dos2unix.exe tensorflow/lite/micro/tools/make/download_and_extract.sh
dos2unix: converting file tensorflow/lite/micro/tools/make/download_and_extract.sh to Unix format...

C:\Versionsverwaltung\tensorflow>C:\MinGW\bin\mingw32-make.exe -f tensorflow/lite/micro/tools/make/Makefile TARGET=esp generate_hello_world_esp_project
process_begin: CreateProcess(NULL, uname -m, ...) failed.
"-m" kann syntaktisch an dieser Stelle nicht verarbeitet werden.
FIND: Parameterformat falsch
FIND: Parameterformat falsch
mingw32-make: *** No rule to make target 'generate_hello_world_esp_project'.  Stop.
```

**Comment by sachinprasadhs** [comment]
The link which you have mentioned does not exists anymore, please follow [this](https://www.tensorflow.org/lite/microcontrollers/get_started_low_level) link and try the `Hello world` example.
Feel free to open new issue if the issue still persists.
